tektoncd/dashboard
# Fix bug where multiple service accounts can't be unpatched at once

# Changes

Different impl, test code needs to be updated in a follow up PR (see https://github.com/tektoncd/dashboard/issues/681), we need this fix in urgently for 0.2.1.

See the big ol' comment in secrets.js, plenty of manual testing done.

We essentially do a replace patch now with a list of known secrets to remain, per service account per namespace (hence lots of for loops).

Testing done:

**Create a, b, c secrets patched to service account tekton-dashboard**
- Sort the table annotation so c, b, a appears.
- Edit the service account manually so it's b, c, a.
- Delete c and a.

**Next, create five secrets.**
- Assign two secrets to one service account `default` in `default`.
- Assign two secrets to one service account, `tekton-dashboard` in tekton-pipelines.
- Assign another secret to a different service account in `tekton-pipelines`.
- Delete ^^ any in any order.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._
